actions/readFile.go

```diff
@@ -0,0 +1,60 @@
+package actions
+
+import (
+	"geektrust/helpers"
+	"geektrust/models"
+	"strings"
+)
+
+//Init function starts the program
+func Init(path string) bool {
+	isFileExist := helpers.FileExists(path)
```

Comment:
How about we edit the `ReadFileFromPath()` function to have this check inbuilt in it? The function can then return a error if there is no file. 

Also, let's [return early](https://dave.cheney.net/practical-go/presentations/qcon-china.html#_return_early_rather_than_nesting_deeply) and avoid nesting when possible. 

```go ****
cmds, er := helpers.ReadFileFromPath(path)
if err != nil || len(cmds) < 0 {
  return false
}
// add logic for when commands are actually present. 
```

Also do you think this function can be better named as `ReadCommandsFromFile(filePath string)`?

Reply:
The check if file does not exist/file is corrupted or it is a malformed path is already done in helpers.FileExists().
Having this check again in ReadCommandsFromFile might add to duplicate code.

Renaming the function name looks good. Will do that.

Reply:
The idea was that we will skip calling the `helpers.FileExists(path)` function here completely once we incorporate that check in the `ReadFileFromPath(path)` . This way `ReadFileFromPath(path)` does not need a prior call to `helpers.FileExists(path)` every time we use it.

Reply:
Even if we want to keep the 2 functions seperate, let's reduce the nesting here by returning early (as suggested above) .
```go
if !isFileExist { 
  return false
}
cmds := helpers.ReadFileFromPath(path)
if len(cmds) <= 0 {
  return false
}
// add logic for when cmds > 0. 
// Notice how we are removing the 'else' here.  
```
Here is a good article: https://dave.cheney.net/practical-go/presentations/qcon-china.html#_return_early_rather_than_nesting_deeply